main.c

```diff
@@ -690,12 +693,21 @@ static void *tcmur_cmdproc_thread(void *arg)
 			tcmu_err("ppoll received unexpected revent: 0x%x\n", pfd.revents);
 			break;
 		}
-	}
 
-	tcmu_err("thread terminating, should never happen\n");
-
-	pthread_cleanup_pop(1);
+		/*
+		 * LIO will wait for outstanding requests and prevent new ones
+		 * from being sent to runner during device removal, but if the
+		 * tcmu cmd_time_out has fired tcmu-runner may still be executing
+		 * requests that LIO has completed. We only need to wait for replies
+		 * for outstanding requests so throttle the cmdproc thread now.
+		 */
+		pthread_mutex_lock(&rdev->state_lock);
+		if (rdev->flags & TCMUR_DEV_FLAG_STOPPING)
+			dev_stopping = true;
+		pthread_mutex_unlock(&rdev->state_lock);
+	}
 
+	pthread_cleanup_pop(0);
```

Comment:
Shoot. This was wrong.

I think we can now hit the race where this completes first, we just do pop(0) and tcmur_cmdproc_thread never gets run leaving the device opened. Will make a new PR for this.